dask/array/tests/test_shuffle.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+import dask
+import dask.array as da
+from dask.array import assert_eq, shuffle
+from dask.core import flatten
+
+
+@pytest.fixture()
+def arr():
+    return np.arange(0, 24).reshape(8, 3).T.copy()
+
+
+@pytest.fixture()
+def darr(arr):
+    return da.from_array(arr, chunks=((2, 1), (4, 4)))
+
+
+@pytest.mark.parametrize(
+    "indexer, chunks",
+    [
+        ([[1, 5, 6], [0, 2, 3, 4, 7]], (3, 5)),
+        ([[1, 5, 6], [0, 3], [4, 2, 7]], (5, 3)),
+        ([[1], [0, 6, 5, 3, 2, 4], [7]], (1, 6, 1)),
+        ([[1, 5, 1, 5, 1, 5], [1, 6, 4, 2, 7]], (6, 5)),
+    ],
+)
+def test_shuffle(arr, darr, indexer, chunks):
+    result = darr.shuffle(indexer, axis=1)
+    expected = arr[:, list(flatten(indexer))]
+    assert_eq(result, expected)
+    assert result.chunks[0] == darr.chunks[0]
+    assert result.chunks[1] == chunks
+
+
+def test_shuffle_config_tolerance(arr, darr):
+    indexer = [[1, 5, 6], [0, 3], [4, 2, 7]]
+    chunks = (3, 2, 3)
+    with dask.config.set({"array.shuffle.chunksize-tolerance": 1}):
+        result = darr.shuffle(indexer, axis=1)
+    expected = arr[:, list(flatten(indexer))]
```

Comment:
nit: I'd personally find this test more readable if we just replaced `list(flatten(indexer))` with the corresponding literal values.

Reply:
sounds good